lib/container-base-object.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const BaseObject = require('./base');
+
+class ContainerBaseObject extends BaseObject {
+  /**
+   * Create generic Kubernetes API object that might contain other resources.
+   * For example, a named Pod contains .log resources (core.ns.pods('foo').log).
+   *
+   * @param {object} options - Options object
+   * @param {string} options.resources - Array of resources to add
+   */
+  constructor(options) {
+    super(options);
+    if (options.resources) {
+      options.resources.forEach(resource => this.addResource(resource));
+    }
+  }
+
+  /**
+   * Add a resource to the container object.
+   * @param {string|object} options - resource name or options object
+   * @param {string} options.name - resource name
+   * @param {fn} options.Constructor - constructor for new resource
+   * @returns {object} returns this to facilitate chaining
+   */
+  addResource(options) {
```

Comment:
Not really. I think it is ok to duplicate if we are going to eventually make `ApiGroup` a `ContainerBaseObject`